index.js

```diff
@@ -50,18 +54,34 @@ Cloneable.prototype.clone = function () {
   // for starting the flow
   this.removeListener('newListener', onData)
   var clone = new Clone(this)
-  this.on('newListener', onData)
+  if (this._hasListener) {
+    this.on('newListener', onData)
+  }
 
   return clone
 }
 
+Cloneable.prototype._destroy = function (err, cb) {
+  if (!err) {
+    this.push(null)
+    this.end()
+    this.emit('close')
```

Comment:
note this will break when node master is released

Reply:
breakage meaning close will be emitted twice

Reply:
We are relying on readable-stream@2 here, and that would end up in readable-stream@3. So it's safe to do it this way atm.

Reply:
:+1: